keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1462,225 +1278,309 @@ prevented. Deleting the entire cluster also leaves resources allocated outside
+    // allocated for the claim.
+    Constraints []Constraint
+
+    // Requests are individual requests for separate resources for the claim.
+    // An empty list is valid and means that the claim can always be allocated
+    // without needing anything. A class can be referenced to use the default
+    // requests from that class.
     //
-    // The object must be in the same namespace as the ResourceClaim.
-    // +optional
-    ParametersRef *ResourceClaimParametersReference
-}
-```
+    // +listType=atomic
+    Requests []Request
 
-The `ResourceClassName` field may be left empty. The parameters are sufficient
-to determine which driver needs to provide resources. This leads to some corner cases:
-- Empty `ResourceClassName` and nil `ParametersRef`: this is a claim which requests
-  no resources. Such a claim can always be allocated with an empty result. Allowing
-  this simplifies code generators which dynamically fill in the resource requests
-  because they are allowed to generate an empty claim.
-- Non-empty `ResourceClassName`, nil `ParametersRef`, nil
-  `ResourceClass.DefaultClaimParametersRef`: this is handled the same way, the
-  only difference is that the cluster admin has decided that such claims need
-  no resources by not providing default parameters.
-
-There is no default ResourceClass. If that is desirable, then it can be
-implemented with a mutating and/or admission webhook.
+    // Future extension, ignored by older schedulers. This is fine because scoring
+    // allows users to define a preference, without making it a hard requirement.
+    //
+    //
+    // Score *SomeScoringStruct
 
-```
-// ResourceClaimStatus tracks whether the resource has been allocated and what
-// the resulting attributes are.
-type ResourceClaimStatus struct {
-    // Allocation is set by the resource driver once a resource or set of
-    // resources has been allocated successfully. If this is not specified, the
-    // resources have not been allocated yet.
+    // Shareable indicates whether the allocated claim is meant to be shareable
+    // by multiple consumers at the same time.
     // +optional
-    Allocation *AllocationResult
+    Shareable bool
+}
 
-    // ReservedFor indicates which entities are currently allowed to use
-    // the claim. A Pod which references a ResourceClaim which is not
-    // reserved for that Pod will not be started.
+// Constraint must have one and only one field set.
+type Constraint struct {
+    // All devices must have this attribute and its value must be the same.
+    //
+    // For example, if you specified "numa.dra.example.com" (a hypothetical example!),
+    // then only devices in the same NUMA node will be chosen.
     //
-    // There can be at most 32 such reservations. This may get increased in
-    // the future, but not reduced.
     // +optional
-    ReservedFor []ResourceClaimConsumerReference
+    // +listType=atomic
+    MatchAttribute *string
```

Comment:
Do we want to make this a list of strings for aesthetic purposes?

Reply:
I don't think so -- this is what I was referring to in my comment here: https://github.com/kubernetes/enhancements/pull/4709#discussion_r1632369182

Reply:
I made `Constraint` a struct with one common field (`RequestNames`) and one-of semantic for the rest (well, the one field we have now, but we know with good confidence that more will come).